src/iceberg_extension.cpp

```diff
@@ -147,25 +150,18 @@ static unique_ptr<Catalog> IcebergCatalogAttach(StorageExtensionInfo *storage_in
 		return std::move(catalog);
 	}
 
-	// Check no endpoint type has been passed.
-	if (!endpoint_type.empty()) {
-		throw IOException("Unrecognized endpoint point: %s. Expected either S3_TABLES or GLUE", endpoint_type);
-	}
-	if (endpoint_type.empty() && endpoint.empty()) {
-		throw IOException("No endpoint type or endpoint provided");
+	catalog_type = ICEBERG_CATALOG_TYPE::POLARIS;
+	credentials.region = region.IsNull() ? "" : region.ToString();
+	if (endpoint.empty()) {
+		Value endpoint_val = kv_secret.TryGetValue("endpoint");
```

Comment:
Re: refactoring our secret functionality. 

It looks like here you get the catalog `endpoint` from the secret if it is not defined in the `Attach` statement. Is there a specific reason you need to get it from the secret?

Reply:
To continue on my comment from above, the fact that you're looking for the catalog endpoint through the  `endpoint` of an s3 secret shows again that this shouldn't be an s3 secret.

See [here](https://duckdb.org/docs/stable/extensions/httpfs/s3api#overview-of-s3-secret-parameters) what the `endpoint` means for an s3 secret.

Reply:
The current plan for the secret rework does not include the ability to provide the `endpoint` in the iceberg secret.. but thinking about it, perhaps this is a good idea.

A single endpoint can provide access to multiple catalogs (or "warehouses"), so putting this into the secret, together with the authorization credentials could be useful for reusability purposes.

Reply:
You are correct, the ENDPOINT doesn't need to live in the secret since it does not need to be a secret, and can be provided as part of the ATTACH command. We can make that as a separate change though if you don't mind

Reply:
It was already in the secret, then it was removed as part of the GLUE/S3_TABLES changes but I don't think they thought through all the implications at that time